api/v1alpha2/conversion.go

```diff
@@ -160,6 +161,7 @@ func (src *MachineSet) ConvertTo(dstRaw conversion.Hub) error {
 
 	if restored.Spec.ClusterName != "" {
 		dst.Spec.ClusterName = restored.Spec.ClusterName
+		dst.Spec.Template.Spec.ClusterName = restored.Spec.ClusterName
```

Comment:
This should not be necessary, should it? This is handled by the call to `restoreMachineSpec` 2 lines down.